include/ews/ews.hpp

```diff
@@ -7602,6 +7602,545 @@ namespace ews
+                sstr << "PropertyType=\"" << property_type_.str() << "\"/>";
+            }
+            return sstr.str();
+        }
+
+        //! Converts an xml string into a extended_field_uri property.
+        static extended_field_uri
+        from_xml_element(const rapidxml::xml_node<>& elem)
+        {
+            extended_field_uri ext_field_uri;
+
+            for (auto attr = elem.first_attribute(); attr != nullptr;
+                 attr = attr->next_attribute())
+            {
+                if (std::string(attr->name(), attr->name_size()) ==
+                    "DistinguishedPropertySetId")
+                {
+                    ext_field_uri.set_distinguished_property_set_id(
+                        distinguished_property_set_id(
+                            std::string(attr->value(), attr->value_size())));
+                }
+                else if (std::string(attr->name(), attr->name_size()) ==
+                         "PropertySetId")
+                {
+                    ext_field_uri.set_property_set_id(property_set_id(
+                        std::string(attr->value(), attr->value_size())));
+                }
+                else if (std::string(attr->name(), attr->name_size()) ==
+                         "PropertyTag")
+                {
+                    ext_field_uri.set_property_tag(property_tag(
+                        std::string(attr->value(), attr->value_size())));
+                }
+                else if (std::string(attr->name(), attr->name_size()) ==
+                         "PropertyName")
+                {
+                    ext_field_uri.set_property_name(
+                        std::string(attr->value(), attr->value_size()));
+                }
+                else if (std::string(attr->name(), attr->name_size()) ==
+                         "PropertyId")
+                {
+                    ext_field_uri.set_property_id(property_id(
+                        std::string(attr->value(), attr->value_size())));
+                }
+                else if (std::string(attr->name(), attr->name_size()) ==
+                         "PropertyType")
+                {
+                    ext_field_uri.set_property_type(property_type(
+                        std::string(attr->value(), attr->value_size())));
+                }
+                else
+                {
+                    throw exception(
+                        "Unexpected child element in <ExtendedFieldURI>");
+                }
+            }
+
+            // Validate and constructor selection
+            if (ext_field_uri.get_property_type().empty())
+            {
+                throw exception(
+                    "Invalid empty <PropertyType> in <ExtendedFieldURI>");
+            }
+            return ext_field_uri;
+        }
+
+        //! Converts a xml string into a extended_field_uri rapidxml
+        //! representation.
+        rapidxml::xml_node<>* to_xml_element(rapidxml::xml_document<char>* doc)
```

Comment:
You expect that `doc` here is never null. You should express that in the signature by taking a reference instead of a pointer.

Also, `to_xml_element` should not just create a new element and return it as it is the case here. It rather should create a new element, append it to a given parent node, **and** return a reference to that newly created element.